pkg/pillar/cmd/zedrouter/metadata.go

```diff
@@ -8,84 +8,13 @@
 package zedrouter
 
 import (
-	"bytes"
-	"encoding/base64"
-	"encoding/binary"
-	"encoding/json"
-	"fmt"
-	"io"
-	"net"
 	"net/http"
-	"os"
-	"path"
-	"strings"
 
+	"github.com/go-chi/chi/v5"
```

Comment:
at some point gorilla was deprecated and had no active maintaners, I heard they found somebody, but better be safe than sorry, plus switching to go-chi costs us nothing since it is based on http.Mux and since we don't have tests for older routes, but it would be nice to have some kind of modern router (because we don't want to parse URL to get route parameters such as /api/v1/{patch_envelope_id}/{file_path}

Reply:
Okay, fine, can you then please remove `gorilla/mux` from `go.{mod,sum}`?

Reply:
We do use it [wstunnelclient](https://github.com/lf-edge/eve/blob/master/pkg/pillar/zedcloud/wstunnelclient.go) so it might be a good conversation to think should we switch to gorilla and how we should switch gorilla websockets

Reply:
IMO we should not use both, because then we basically double the risk that one project gets deprecated and we have to fix it and you basically said "better be safe than sorry".

(Ideally we abstract it away from the underlying muxer so we have to change it only at one place, but that's too much for now, I think)

Reply:
@uncleDecart Do we need anything fancy from the router necessating switching from the basic and standard http.NewServeMux?

Reply:
Reason why I'm bringing go-chi here is because I want to use path parameters which, i.e. 
`/eve/v1/patch/{patch-id}/{file-name}`
It is not yet it PR, I'm implementing logic. We, of course can write our own parser, but I'm trying to avoid reinventing wheel here. Another reason I'm using go-chi is because it's 100% compatible with `http/net` and has no other dependencies except this package.

Reply:
Makes sense.

Reply:
So @christoph-zededa we do not use gorilla mux explicitly in EVE, however it's indirect dependency from gorilla ws, which is not EOL and also used to communicate with controller and in controller we are still using gorilla ws, so I suggest we keep using gorilla ws for now